pipeline_tools/get_analysis_metadata.py

```diff
@@ -0,0 +1,48 @@
+import argparse
+import json
+import requests
+from requests.auth import HTTPBasicAuth
+from tenacity import retry, wait_exponential, stop_after_delay
+
+
+def get_workflow_id(analysis_output_path):
+    url = analysis_output_path
+    hash_end = url.rfind("/call-")
+    hash_start = url.rfind('/', 0, hash_end) + 1
+    workflow_id = url[hash_start:hash_end]
+    # write workflow_id to a file so it is available outside of the get_analysis task
+    with open('workflow_id.txt', 'w') as f:
+        f.write(workflow_id)
+    return workflow_id
+
+
+def get_auth(credentials_file=None):
+    credentials_file = credentials_file or '/cromwell-metadata/cromwell_credentials.txt'
+    with open(credentials_file) as f:
+        credentials = f.read().split()
+    user = credentials[0]
+    password = credentials[1]
+    return HTTPBasicAuth(user, password)
+
+
+@retry(wait=wait_exponential(multiplier=1, max=10), stop=stop_after_delay(20))
+def get_metadata(runtime_environment, workflow_id):
+    print('Getting metadata for workflow {}'.format(workflow_id))
+    url = 'https://cromwell.mint-{}.broadinstitute.org/api/workflows/v1/{}/metadata?expandSubWorkflows=true'.format(runtime_environment, workflow_id)
+    response = requests.get(url, auth=get_auth())
+    response.raise_for_status()
+    with open('metadata.json', 'w') as f:
+        json.dump(response.json(), f, indent=2)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--analysis_output_path', required=True)
+    parser.add_argument('--runtime_environment', required=True)
+    args = parser.parse_args()
+    workflow_id = get_workflow_id(args.analysis_output_path)
+    get_metadata(args.runtime_environment, workflow_id)
```

Comment:
I personally prefer to use something like `get_metadata.retry_with(wait=wait_exponential(multiplier=1, max=10), stop=stop_after_delay(20))(args.runtime_environment, workflow_id)` here explicitly so that we could control the retry parameters without going back to change the decorator parameters. But this is totally **OPTIONAL**

Reply:
Ok! I actually prefer the decorator pattern, but I think this would be useful if there are multiple requests we want to retry in the same function

Reply:
Oh, or to pass in the retry parameters to the function which I think is what you meant originally! Good idea!

Reply:
Oh yeah, that is I originally meant! You read my mind LOL